driver/src/main/java/com/code_intelligence/jazzer/driver/FuzzTargetFinder.java

```diff
@@ -0,0 +1,193 @@
+            .collect(Collectors.toList());
+    if (annotatedMethods.isEmpty()) {
+      return Optional.empty();
+    }
+
+    if (annotatedMethods.size() > 1) {
+      throw new IllegalArgumentException(String.format(
+          "Each class may only contain a single method annotated with @FuzzTest, got multiple in %s",
+          clazz.getName()));
+    }
+
+    // The following checks ensure compatibility with the JUnit concept of a test class and test
+    // method.
+    // https://junit.org/junit5/docs/5.9.0/user-guide/#writing-tests-definitions
+    Method method = annotatedMethods.get(0);
+    if (Modifier.isPrivate(method.getModifiers())) {
+      throw new IllegalArgumentException(
+          String.format("Methods annotated with @FuzzTest must not be private, got %s in %s",
+              method.getName(), clazz.getName()));
+    }
+    if (Modifier.isStatic(method.getModifiers())) {
+      throw new IllegalArgumentException(
+          String.format("Methods annotated with @FuzzTest must not be static, got %s in %s",
+              method.getName(), clazz.getName()));
+    }
+    if (method.getParameterCount() > 1) {
+      throw new IllegalArgumentException(String.format(
+          "Methods annotated with @FuzzTest must take a single parameter, got %d for %s in %s",
+          method.getParameterCount(), method.getName(), clazz.getName()));
+    }
+
+    Class<?> parameter = method.getParameterTypes()[0];
+    if (parameter != byte[].class && parameter != FuzzedDataProvider.class) {
+      throw new IllegalArgumentException(String.format(
+          "Methods annotated with @FuzzTest must take a single byte[] or FuzzedDataProvider parameter, got %s for %s in %s",
+          parameter.getName(), method.getName(), clazz.getName()));
+    }
+
+    if (clazz.getDeclaredConstructors().length != 1) {
+      throw new IllegalArgumentException(String.format(
+          "Classes containing a method annotated with @FuzzTest must declare exactly one constructor, got multiple in %s",
+          clazz.getName()));
+    }
+    Constructor<?> constructor = clazz.getDeclaredConstructors()[0];
+    // JUnit 5 supports injected constructor parameters, but we don't.
+    if (constructor.getParameterCount() > 0) {
+      throw new IllegalArgumentException(String.format(
+          "The constructor of a class containing a method annotated with @FuzzTest must take no parameters, got a non-zero number in %s",
+          clazz.getName()));
+    }
+
+    // Both the constructor and the method may not be accessible - JUnit test classes and methods
+    // are usually declared without access modifiers and thus package-private.
+    method.setAccessible(true);
+    constructor.setAccessible(true);
+
+    return Optional.of(new FuzzTarget(parameter == FuzzedDataProvider.class, method,
+        constructor::newInstance, targetPublicStaticMethod(clazz, FUZZER_TEAR_DOWN)));
```

Comment:
Mixing annotation based configuration and "name based" configuration with `fuzzerTearDown` feels strange. Should we skip the tear down function in this case, or use `@AfterAll` (multiple methods could be annotated)?

Reply:
Agreed, we should use JUnit concepts here in the future. I have rarely seen uses for `fuzzerTearDown` (if any at all?), so I will leave it out for now.

Reply:
By the way, shouldn't `@BeforeXX` and `AfterXX` work out of the box?

Reply:
They work out of the box for the regression test, but not if Jazzer executes the fuzz test - after all `@Before/AfterXXX` is part of the Jupiter engine, not the JUnit platform.